src/main/java/org/broadinstitute/hellbender/tools/spark/BaseRecalibratorSpark.java

```diff
@@ -90,7 +90,7 @@ protected void runTool( JavaSparkContext ctx ) {
 
         // TODO: Look into broadcasting the reference to all of the workers. This would make AddContextDataToReadSpark
         // TODO: and ApplyBQSRStub simpler (#855).
-        JavaPairRDD<GATKRead, ReadContextData> rddReadContext = AddContextDataToReadSpark.add(ctx, initialReads, getReference(), bqsrKnownVariants, joinStrategy, getReferenceSequenceDictionary(), readShardSize, readShardPadding);
+        JavaPairRDD<GATKRead, ReadContextData> rddReadContext = AddContextDataToReadSpark.add(ctx, initialReads, getReference(), bqsrKnownVariants, knownVariants, joinStrategy, getHeaderForReads().getSequenceDictionary(), readShardSize, readShardPadding);
```

Comment:
The RDD of variants is not used anymore when joinStrategy == OVERLAPS_PARTITIONER, is it? If it's not needed, can you conditionally avoid the cost of loading the variants RDD by checking the join strategy requested? Applies to other clients of `AddContextDataToReadSpark` as well.